src/Type/Php/ArrayFilterFunctionReturnTypeExtension.php

```diff
@@ -78,64 +84,59 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		if ($flagArg === null) {
-			if ($callbackArg instanceof Closure && count($callbackArg->stmts) === 1 && count($callbackArg->params) > 0) {
-				$statement = $callbackArg->stmts[0];
-				if ($statement instanceof Return_ && $statement->expr !== null) {
-					return $this->filterByTruthyValue($scope, $callbackArg->params[0]->var, $arrayArgType, null, $statement->expr);
-				}
-			} elseif ($callbackArg instanceof ArrowFunction && count($callbackArg->params) > 0) {
-				return $this->filterByTruthyValue($scope, $callbackArg->params[0]->var, $arrayArgType, null, $callbackArg->expr);
-			} elseif ($callbackArg instanceof String_) {
-				$funcName = self::createFunctionName($callbackArg->value);
-				if ($funcName === null) {
-					return new ErrorType();
-				}
-
-				$itemVar = new Variable('item');
-				$expr = new FuncCall($funcName, [new Arg($itemVar)]);
-				return $this->filterByTruthyValue($scope, $itemVar, $arrayArgType, null, $expr);
-			}
+			$mode = self::USE_ITEM;
+		} elseif ($flagArg instanceof ConstFetch && $flagArg->name->getParts()[0] === 'ARRAY_FILTER_USE_KEY') {
```

Comment:
Should use `scope->getType()` instead of raw-Ast, or even https://github.com/phpstan/phpstan-src/blob/1.12.x/src/Type/BitwiseFlagHelper.php

Reply:
Fixed.
I borrowed `getConstant` method from `FilterFunctionReturnTypeHelper` and `PathinfoFunctionDynamicReturnTypeExtension`.
Probably it is better to move them into a helper class, but it can be done in another PR.